### PR TITLE
[VCDA-4355] Update VcdProperties

### PR DESCRIPTION
**Pull request by Anirudh9794**
## Description
Please provide a brief description of the changes proposed in this Pull Request

- Make org and vdc an array in vcdProperties
- update external ID of the RDE

## Checklist
- [ ] tested locally
- [x] updated any relevant dependencies
- [x] updated any relevant documentation or examples

## API Changes
Are there API changes?
- [ ] Yes
- [x] No

If yes, please fill in the below

1. Updated conversions?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
2. Updated CRDs?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
3. Updated infrastructure-components.yaml?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
4. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [ ] No
    - [ ] N/A
5. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [ ] Yes
   - [ ] No
   - [ ] N/A

## Issue
If applicable, please reference the relevant issue

Fixes #

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251)

### CODE REVIEW

**Review by sahithi — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251)* status: all files reviewed, 4 unresolved discussions (waiting on @Anirudh9794)

___
*[`controllers/vcdcluster_controller.go` line 210 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-NBqt2VQC1nMUoxNFXBo:-NBqt2VQC1nMUoxNFXBp:b9l2pmz) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/7e87f54abe79b8764ef17d925e1b820c2edcf978/controllers/vcdcluster_controller.go#L210)):*
> ```Go
> 	}
> 	capvcdEntity := rdeType.CAPVCDEntity{
> 		ExternalID: "",
> ```

ExternalId is the property of RDE object, not CAPVCDEntity.
___
*[`pkg/capisdk/defined_entity.go` line 216 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-NBqtq9aEdjcZFFL-UwV:-NBqtq9aEdjcZFFL-UwW:b-glwcvb) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/7e87f54abe79b8764ef17d925e1b820c2edcf978/pkg/capisdk/defined_entity.go#L216)):*
> ```Go
> 		}
> 		if updateExternalID {
> 			klog.V(4).Infof("setting externalID as [%s]", externalID)
> ```

Add rdeId to the log statement.
___
*[`pkg/vcdtypes/rde_type_1_1_0/types_1_1_0.go` line 158 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-NBqu0Y97AsfHmC-C87j:-NBqu0Y97AsfHmC-C87k:b-jd2uof) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/7e87f54abe79b8764ef17d925e1b820c2edcf978/pkg/vcdtypes/rde_type_1_1_0/types_1_1_0.go#L158)):*
> ```Go
> 	Status     Status     `json:"status"`
> 	Kind       string     `json:"kind"`
> 	ExternalID string     `json:"externalId,omitempty"`
> ```

ExternalId is not part of the entity. It sits outside the entity



_Code snippet:_
```Go
RDE {
    Id : RDEId
    externalId: vApp Id
    name: clustername
    entity: { 
         spec:
         status
    }
    state : RESOLVED
}
```
___
*[`pkg/vcdtypes/rde_type_1_1_0/types_1_1_0.go` line 159 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-NBquaOl4lVCo6IMI09G:-NBquaOl4lVCo6IMI09H:bqqe7y1) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/7e87f54abe79b8764ef17d925e1b820c2edcf978/pkg/vcdtypes/rde_type_1_1_0/types_1_1_0.go#L159)):*
> RDE {
<details><summary><i alt="summary">Quoted 8 lines of code…</i></summary>

>     Id : RDEId
>     externalId: vApp Id
>     name: clustername
>     entity: {·
>          spec:
>          status
>     }
>     state : RESOLVED
</details>
> }

**Review by sahithi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251)* status: all files reviewed, 5 unresolved discussions (waiting on @Anirudh9794)

___
*[`examples/rde1_1_0.json` line 22 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-NBqujVR0QU6woRxxpFj:-NBqujVR0QU6woRxxpFk:b2lt0oz) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/7e87f54abe79b8764ef17d925e1b820c2edcf978/examples/rde1_1_0.json#L22)):*
> ```JSON
>   },
>   "apiVersion": "ABCDEFG",
>   "externalId": "VAppID",
> ```

As mentioned above, this is not part of our schema

**Review by Anirudh9794 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251)* status: all files reviewed, 5 unresolved discussions (waiting on @Anirudh9794 and @sahithi)

___
*[`controllers/vcdcluster_controller.go` line 210 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-NBqt2VQC1nMUoxNFXBo:-NBrdm8Q6F9fFrzFmYK2:b-896fix) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/7e87f54abe79b8764ef17d925e1b820c2edcf978/controllers/vcdcluster_controller.go#L210)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

ExternalId is the property of RDE object, not CAPVCDEntity.
</blockquote></details>

Done.
___
*[`pkg/capisdk/defined_entity.go` line 216 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-NBqtq9aEdjcZFFL-UwV:-NBreX_SAC3UNwA0qzDy:b-896fix) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/7e87f54abe79b8764ef17d925e1b820c2edcf978/pkg/capisdk/defined_entity.go#L216)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

Add rdeId to the log statement.
</blockquote></details>

Done.
___
*[`pkg/vcdtypes/rde_type_1_1_0/types_1_1_0.go` line 159 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-NBquaOl4lVCo6IMI09G:-NBredbt-5YsO4QaeXwE:bma5s22) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/7e87f54abe79b8764ef17d925e1b820c2edcf978/pkg/vcdtypes/rde_type_1_1_0/types_1_1_0.go#L159)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

> RDE {
>     Id : RDEId
>     externalId: vApp Id
>     name: clustername
>     entity: {·
>          spec:
>          status
>     }
>     state : RESOLVED
> }

</blockquote></details>

Removed the entity ID from here

**Review by sahithi — APPROVED** [review]
:lgtm: 

> Reviewed 2 of 6 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-:-NBsQfUx-50SzGMm0n3j:bg7igkv)* status: all files reviewed, 5 unresolved discussions (waiting on @Anirudh9794 and @sahithi)

**Review by lzichong — APPROVED** [review]
:lgtm:, will update CAPVCD Schema after merge

> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/251#-:-NBsWNzS9f0OxY0oqnRp:bnank6a)* status: all files reviewed, 5 unresolved discussions (waiting on @Anirudh9794 and @sahithi)

**Pull request merged**
